features/apimgt/org.wso2.carbon.apimgt.admin.feature/src/main/resources/admin/source/src/app/components/GatewayEnvironments/AddEditGWEnvironment.jsx

```diff
@@ -115,30 +146,41 @@ function AddEditMGLabel(props) {
                 }
                 if (value === '') {
                     error = 'Name is Empty';
-                } else if (/\s/.test(value)) {
-                    error = 'Name contains spaces';
                 } else if (/[!@#$%^&*(),?"{}[\]|<>\t\n]/i.test(value)) {
                     error = 'Name field contains special characters';
                 } else {
                     error = false;
                 }
                 break;
-            case 'hosts':
-                if (hosts === undefined) {
+            case 'displayName':
+                if (value === '') {
+                    error = 'Display Name is Empty';
+                } else {
+                    error = false;
+                }
+                break;
+            case 'vhosts': {
+                if (value === undefined) {
```

Comment:
I think this is equitant to 

```suggestion
                if (!value) {
```